packages/rich-text/src/ContentfulEditorProvider.ts

```diff
@@ -9,15 +10,24 @@ export function getContentfulEditorId(sdk: FieldExtensionSDK) {
   return `rich-text-editor-${sys.id}-${field.id}-${field.locale}`;
 }
 
-interface useContentfulEditorHookProps {
-  sdk: FieldExtensionSDK;
+export const editorContext = createContext('');
+
+export const ContentfulEditorIdProvider = editorContext.Provider;
+
+export function useContentfulEditorId() {
+  const id = useContext(editorContext);
+  if (!id) {
+    throw new Error(
+      'could not find editor id. Please ensure the component is wrapped in <ContentfulEditorIdProvider> '
+    );
+  }
+
+  return id;
 }
 
-function useContentfulEditorHook({ sdk }: useContentfulEditorHookProps) {
-  const editorId = getContentfulEditorId(sdk);
-  const editor = usePlateEditorRef(editorId);
+export function useContentfulEditor() {
+  const editorId = useContentfulEditorId();
+  const editor = usePlateEditorState(editorId);
```

Comment:
Wondering how did it work in the past? well, because `onChange` wasn't memoized that caused the creation of a new `editor` instance on every change :)